node/impl/full/chain.go

```diff
@@ -557,21 +559,16 @@ func (a *ChainAPI) ChainGetNode(ctx context.Context, p string) (*api.IpldObject,
 
 	bs := a.ExposedBlockstore
 	bsvc := blockservice.New(bs, offline.Exchange(bs))
+	fc := bsfetcher.NewFetcherConfig(bsvc)
+	r := resolver.NewBasicResolver(fc)
 
-	dag := merkledag.NewDAGService(bsvc)
-
-	r := &resolver.Resolver{
-		DAG:         dag,
-		ResolveOnce: resolveOnce(bs, a.TsExec),
-	}
-
-	node, err := r.ResolvePath(ctx, ip)
+	node, lnk, err := r.ResolvePath(ctx, ip)
```

Comment:
This needs to use the `resolveOnce` we implement above; It's what implements most of the functionality of this API